File: official/docs/java/v6/smartrate/retrieve-estimated-delivery-date.java
package shipments;

import com.easypost.exception.EasyPostException;
import com.easypost.model.Shipment;
import com.easypost.model.EstimatedDeliveryDate;
import com.easypost.service.EasyPostClient;

import java.util.List;

public class RetrieveEstimatedDeliveryDate {
    public static void main(String[] args) throws EasyPostException {
        EasyPostClient client = new EasyPostClient("EASYPOST_API_KEY");

        Shipment shipment = client.shipment.retrieve("shp_...");
        List<EstimatedDeliveryDate> estimatedDeliveryDates = client
                .shipmentretrieveEstimatedDeliveryDate(shipment.getId(), "YYYY-MM-DD");

        System.out.println(estimatedDeliveryDates);
    }
}

File: official/docs/ruby/v6/smartrate/retrieve-estimated-delivery-date.rb
require 'easypost'

client = EasyPost::Client.new(api_key: 'EASYPOST_API_KEY')

estimated_delivery_dates = client.shipment.retrieve_estimated_delivery_date(
  'shp_...',
  'YYYY-MM-DD',
)

puts estimated_delivery_dates
